src/main/java/org/candlepin/controller/EntitlementCertificateGenerator.java

```diff
@@ -328,6 +333,122 @@ public void regenerateCertificatesOf(Consumer consumer, boolean lazy) {
         this.regenerateCertificatesOf(consumer.getEntitlements(), lazy);
     }
 
+    @Transactional
+    public void regenerateCertificatesOf(Consumer consumer, List<String> updatedEnvironments,
+        List<String> environmentChangeList, boolean lazy) {
+        log.info(
+            "Regenerating #{}, entitlement certificates for consumer: {}",
+            consumer.getEntitlements().size(), consumer
+        );
+
+        // we need to clear the content access cert on regenerate
+        Owner owner = ownerCurator.findOwnerById(consumer.getOwnerId());
+        if (owner.isUsingSimpleContentAccess()) {
+            this.contentAccessManager.removeContentAccessCert(consumer);
+        }
+
+        List<Entitlement> entitlements = filterEntitlements(consumer,
+            updatedEnvironments, environmentChangeList);
+
+        // TODO - Assumes only 1 entitlement certificate exists per entitlement
+        this.regenerateCertificatesOf(entitlements, lazy);
+    }
+
+    /**
+     * Select consumers entitlement whose entitlement certs need to be
+     * regenerated.
```

Comment:
Let's be more descriptive here and mention the `why` we only want to regenerate some entitlements and not others